### PR TITLE
feat: a pause that cuts a French sentence in two is repaired

### PR DESCRIPTION
**Pull request by znat**
A pause in the middle of a sentence makes the transcriber end it early. The
`interpret` step repairs that in English. This turns it on for French, and
fixes two reasons an English capital left over from a pause was not touched.

French reads at AUC 0.968 against 0.984 in English, repairs 91% of cuts, and
costs about 2 false joins per 100 real periods. Nothing is behind a setting: a
French dictation that runs `interpret` now gets its boundaries read. The
bare-capital half stays English, because its refusal rules were tuned on
English capitalisation.

Start the review at the false-join rate. English aims at zero and French is at
two per hundred, which is the one number here that is a judgement call rather
than a measurement.

<details>
<summary>French was refused for a reason that expired when ModernBERT left</summary>

`SentenceJoin.apply` held `guard language == "en"`, and the comment above it
gave two reasons. The first was that the readings are scored by an English base
model. That model is gone: the stage reads `Qwen3-0.6B-Base` through
`SentenceReadings`, and `RetiredModels.swift` exists to delete ModernBERT's
cache from machines that still hold it. Nobody revisited the guard when the
model was swapped.

The second reason, that the mark set is English, was real but small.
`Config.Language.builtIn` already had an `"fr"` entry carrying English's marks.

Three things had to change, not one:

- the guard, in the stage **and again in `SentenceJoinCommand`**, so
  `--sentence-join` answers in French too;
- the space French sets before `?`. `boundaries(in:scanning:)` took the
  letters immediately before the mark, so `travailler ?` yielded no boundary at
  all. Removing the guard alone would have read French periods and skipped every
  French question, silently;
- `:` in the French mark set, which French writes where English writes a full
  stop. It is a reading, not an ender.

</details>

<details>
<summary>French, measured on 234 boundaries — AUC 0.968, 91% of cuts repaired</summary>

103 real periods mined from real dictation and 131 cuts made by inserting
`. Majuscule` mid-clause, the same method the English work used.

    AUC (join margin separates cut from real)   0.968
    synthetic cuts repaired                     119/131   91%
    false joins per 100 real periods            5.8 raw, about 2 hand-checked
    median latency                              53 ms, same as English

For comparison: Qwen English 0.984, mmBERT French 0.918, ModernBERT French
0.870, XLM-R French 0.881.

The raw false-join column is overstated, the same way it was in English. All
six were read by hand: two are genuine — `…devrait réapparaître. | Après, par
exemple` and `…en dessous. | Des options plutôt que` — and the other four are
cuts the model repaired correctly, which the mining had labelled as real
periods.

**The comma decoy transfers.** 43 of 103 real periods pick the comma, against
26% in English, and only 6 pick join. That is the mechanism working in a
language it was never tuned on.

The colon changes no decision on the 234. It wins once, on a real period, where
it replaces a comma win. It can only cost a repair, never add a false join.

Earlier work put the corpus ceiling at 51 real periods. It is gone: the
transcript history holds 584 distinct French dictations.

</details>

<details>
<summary>A quantifier after a pause was refused before the model ever saw it</summary>

`…for portrait examples containing Several versions of the same` is a real
dictation, with 2.8s of silence before `Several`. It was never read.

`SentenceJoin.readable` allowed eight lexical classes and Adjective was not one,
because the class is mixed. Measured with `NLTagger`:

    quantifiers tagged Adjective   Several Many Most Few Other Such Same Half
                                   Various Certain Next Last First
    names tagged Adjective         French, Open

**The lemma separates them.** It keeps the capital on a name — `French`→French,
`English`→English, `Google`→Google, `Friday`→Friday — and drops it on a
quantifier — `Several`→several. So an adjective whose lemma starts lowercase may
now be read.

`readableClasses` is untouched, so the Noun refusal that carries most of the
measured 93% is intact. `Slack` lemmatises lowercase too, but it tags Noun and
never reaches the test.

Nine cases added to `tests/sentence-case-cases.json`: five quantifiers and four
names. 48/48, and the 39 that were there do not move.

</details>

<details>
<summary>On a bare capital the comma could only veto — dropping it gains 40 repairs for 2</summary>

A capital with no mark in front of it got four readings. They split two-two on
the only question that path asks:

    .            …then. The dot is wrong      capital kept
    as-decoded   …then The dot is wrong       capital kept
    ,            …then, the dot is wrong      lowercased
    join         …then the dot is wrong       lowercased

Only `join` writes. So a comma win was never a decision. It was a veto on a join
that had already beaten both readings that keep the capital.

Measured over 616 bare capitals mined from 3,336 English dictations, 225 of
which get past the part-of-speech filter:

    lowercased, of 225
      before                87
      after                129

The change acts on 42 cases. 40 are right and **2 destroy a correct capital** —
`…after several retries it fails | The ingest should stop`, and
`In Google just ask quote | What are the top ten questions…`, where the capital
opens a quotation. Both are marked `cost: true` in the fixture and printed by
the check, not hidden.

The marked path keeps its comma. 26% of real sentence endings pick it there,
which is why none of them picks join.

</details>

<details>
<summary>A margin threshold was tried on both, and loses on both</summary>

The two destroyed capitals are thin, +0.061 and +0.056. But so are many repairs:
17 of the 40 sit below +0.10 and the smallest is +0.004.

    threshold   repairs kept   destructions
      0.000        40/40            2
      0.040        30/40            2
      0.062        27/40            0
      0.150        17/40            0

Buying back those two costs 13 repairs. It also reintroduces the calibrated
constant `SentenceJoin.Tier` refuses: "an argmax has no threshold to hedge
with". So the change ships plain.

Grouping `{., as-decoded}` against `{,, join}` was also tried, and is worse: it
repairs 53 of 59 on a labelled sample but destroys 3, including two clean
sentence starts where only the comma was voting.

</details>

<details>
<summary>What this does not fix</summary>

- **The bare-capital half is English only.** Its refusal rules were tuned on
  English capitalisation and French capitalises far less — no days, months,
  nationalities. Untested in French, so it is gated off there.
- **A marked boundary the model reads as a comma still writes nothing.**
  `…the selected text. And also write` loses at `,` -4.05 against `join` -4.43.
  That is a model error on a hard case, not something a rule can reinterpret.
- **The French false-join rate is about 2 per 100**, against English's target of
  zero. One case is in the fixture as a recorded cost.
- **Synthetic cuts are easier than real ones.** They are inserted at a random
  word boundary; real spurious cuts fall at natural pause points. 91% is
  probably optimistic.
- The bare-capital labels are single-annotator, read from the text before any
  score was seen.

</details>

<details>
<summary>Review notes — 54 lines of behaviour, the rest is evidence</summary>

    Sources/ParrotFlow/SentenceJoin.swift          37   languages, the ? space,
                                                        readable()'s lemma clause,
                                                        the guard, the English gate
    Sources/ParrotFlow/SentenceReadings.swift      13   bareReadings drops the comma
    Sources/ParrotFlow/SentenceProbeCommand.swift   5   the bench reads row.language
    Sources/ParrotFlow/Config.swift                 4   ":" in the fr marks
    Sources/ParrotFlow/SentenceJoinCommand.swift    2   the second guard

Everything else is two check scripts, two fixtures and the docs.

`--sentence-probe --bench` now reads the mark set for each row's `language`, so
the French set is reproducible. It defaults to English, so existing case files
are unaffected.

Four commits, each self-contained, in the order the work was done.

Checks, all needing the 320 MB sentence model except `sentence-case`:

    scripts/check-sentence-case.sh       48/48
    scripts/check-sentence-window.sh     11/11
    scripts/check-bare-capital.sh        43/43  and 2 known costs
    scripts/check-french-boundary.sh     27/27  and 1 known cost
    make test                            93 passed, 0 failed

`tests/french-boundary-cases.json` is a 28-case subset of the 234 with names
filtered out.

</details>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/ParrotFlow/Config.swift [hunk 1]
```diff
@@ -1729,7 +1729,9 @@ struct Config: Decodable, Equatable {
             /// wrong rewrites it catches.
             static let builtIn: [String: Language] = [
                 "en": Language(marks: defaultMarks, slotFloor: defaultSlotFloor),
-                "fr": Language(marks: defaultMarks, slotFloor: 0.30),
+                // French writes a colon where English writes a full stop, so
+                // it is a reading here and not an ender.
+                "fr": Language(marks: [".", ",", "?", ":"], slotFloor: 0.30),
```
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/Config.swift:1734]: Declined, and the PR text was wrong rather than the code.

`!` is in no mark set. `defaultMarks` is `[".", ",", "?"]`, so English does not
scan exclamation boundaries either. Adding `!` to French alone would make the
two languages disagree, and adding it to both is a behaviour change nobody has
measured — the 234-case French set is periods only, and `;` and `:` were the
last marks measured in English, where neither changed a decision.

`!` is in `SentenceReadings.enders`, which is a different list: it says the word
after the mark keeps its capital, not that the mark opens a boundary.

So the code stands and the wording goes. The source comment and the docs said
"the space French sets before `?` and `!`"; they now say `?` only, which is what
is actually scanned. Fixed in 15670d20.

File: Sources/ParrotFlow/SentenceJoin.swift [hunk 2]
```diff
@@ -126,7 +141,11 @@ actor SentenceJoin {
             guard at > text.startIndex, cursor < text.endIndex, text[cursor] != mark else {
                 continue
             }
-            let word = text[..<at].reversed().prefix { $0.isLetter || $0.isNumber }
+            // French sets a space before `?` and `!` — "travailler ?" — so the
+            // word is looked for past it. Without this every French question
+            // boundary is invisible and nothing says so.
+            let ahead = text[..<at].reversed().drop { $0.isWhitespace }
+            let word = ahead.prefix { $0.isLetter || $0.isNumber }
```
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/SentenceJoin.swift:156]: Confirmed and fixed in 15670d20. Reproduced first:

    said   Je me demande si on ne devrait pas ? Attendre la prochaine version.
    was    ...on ne devrait pas··Attendre la prochaine version.
    now    Je me demande si on ne devrait pas attendre la prochaine version.

With a mark the leading space goes with it and the separator after it is kept.
With no mark there is nothing after the boundary, so the space in the head is
the only one and it stays — trimming there would run the two words together.

Chasing it found a second bug you did not flag, in the same output. Note the
capital above: `Attendre` survived the join. `written()` built its `NLTagger`
with `.setLanguage(.english)` whatever the dictation, and NLTagger gives no
lemma for most French words. "No lemma" is exactly what that rule reads as a
name, so a joined French boundary kept its capital — `demain. Attendre` became
`demain Attendre`. The tagger now reads the language of the dictation.

Neither was visible to the bench: `--sentence-probe --bench` scores which
reading wins and never applies one. `scripts/check-french-boundary.sh` now
asserts the written text as well, with the case above as the regression.

File: Sources/ParrotFlow/Config.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/Config.swift:1734]: Correction to my reply above: when I posted it the docs were fixed but the
source comment was not, so "fixed in 15670d20" was wrong for half of it.
`SentenceJoin.swift:151` still said "a space before `?` and `!`". It now says
`?`, and adds that `!` is in no mark set in either language, so it is not
scanned here. Done in fdf5e518. The PR body carried the same overclaim and has
been edited.
